feat: Release `1.0.0-pre.9` (#3579)
CloudFire-LLC/cloudfire: website/src/app/kb/authenticate/oidc/readme.mdx

import Alert from "@/components/DocsAlert";
import PlanBadge from "@/components/PlanBadge";
import SupportOptions from "@/components/SupportOptions";

<PlanBadge plans={["starter", "enterprise"]}>

# SSO with OpenID Connect

</PlanBadge>

Firezone supports authenticating users with a universal OIDC connector that
works with any authentication service offering a standard
[OIDC](https://en.wikipedia.org/wiki/OpenID) authentication mechanism. Use this
connector to enable authenticating users and admins to Firezone for any
OIDC-capable identity provider that supports the `authorization_code` grant.

This connector works great for popular hosted providers like Google Workspace,
Microsoft Entra ID, and Okta and also for self-hosted ones like
[Keycloak](https://www.keycloak.org) and [Ory](https://www.ory.sh).

<Alert color="warning">
  User / group sync is **not supported** with the universal OIDC connector. See
  the [Google Workspace](/kb/authenticate/google), [Microsoft Entra
  ID](/kb/authenticate/entra), or [Okta](/kb/authenticate/okta) connectors for

>   automatic user / group sync.

Reviewer: Just a nitpick: this way we are not repetitive and have an extra keyword for search indexers.
```suggestion
  automatic directory sync.
```

<PlanBadge plans={["starter", "enterprise"]}>

# SSO with OpenID Connect

</PlanBadge>

Firezone supports authenticating users with a universal OIDC connector that
works with any authentication service offering a standard
[OIDC](https://en.wikipedia.org/wiki/OpenID) authentication mechanism. Use this
connector to enable authenticating users and admins to Firezone for any
OIDC-capable identity provider that supports the `authorization_code` grant.

This connector works great for popular hosted providers like Google Workspace,
Microsoft Entra ID, and Okta and also for self-hosted ones like
[Keycloak](https://www.keycloak.org) and [Ory](https://www.ory.sh).

<Alert color="warning">
  Directory sync is **not supported** with the universal OIDC connector. See the
  [Google Workspace](/kb/authenticate/google), [Microsoft Entra
  ID](/kb/authenticate/entra), or [Okta](/kb/authenticate/okta) connectors for
  automatic directory sync. You'll need to manually create and manage users and
  groups for use with the universal OIDC connector.
</Alert>

For Firezone-specific instructions for a given provider, select your provider in
the list below:

- [Fusion Auth](/kb/authenticate/oidc/fusion)

> Fo others, consult your provider's documentation for setting up an OpenID

Reviewer 2: ```suggestion
For all other identity providers, consult your provider's documentation for setting up an OpenID
```

- [Keycloak](https://www.keycloak.org/docs/latest/securing_apps/index.html#_oidc)
- [Ory](https://www.ory.sh/docs/getting-started/ory-network-oauth2#authorization-code-grant)
- [Authentik](https://www.ory.sh/docs/oauth2-oidc/authorization-code-f)

## Setting up the universal OIDC connector

To set up the universal OIDC connector, go to `Settings` -> `Identity Providers`
-> `Add Identity Provider` and select `OpenID Connect` as the identity provider.

In general, you'll need three pieces of information to set up the connector:

- [Scopes](#scopes): These control what information Firezone can access from
  your identity provider. At a minimum, you'll need to provide the `openid`,
  `profile`, and `email` scopes. These are configured in your identity
  provider's OAuth app settings.
- [Redirect URIs](#redirect-uris): These are unique to each provider in your
  Firezone account and are used to complete the authentication process. These
  are configured in your identity provider's OAuth app settings.
- [Client ID and secret](#client-id-and-secret): These are used to authenticate

>   Firezone with your identity provider. These are configured in Firezone.

Reviewer 2: ```suggestion
  Firezone with your identity provider. These are generated in your identity provider and then configured in Firezone.
```

- [Discovery document URI](#discovery-document-uri): This is the URL to your
  identity provider's OIDC discovery document. This is used to automatically

>   configure the connector with your identity provider's settings and is

Reviewer 2: ```suggestion
  configure the connector with your identity provider's settings. The URL is generated by your identity provider and then
```

  configured in Firezone.

### Scopes

Firezone requires the following scopes to be added on the connector at a
minimum:

- `openid`: Required by all OpenID Connect integrations and used to identity
  this user in Firezone
- `profile`: Required for providing the user's name
- `email`: Required for authentication

### Redirect URIs

When setting up the connector, you'll need to provide two redirect URIs in the
connector's allowlist. These are shown in the setup form and are unique to each
provider in your Firezone account. They allow Firezone to receive authentication
tokens from your identity provider to complete the authentication process.

> ### Client ID and secret

Reviewer 2: ```suggestion
### Client ID and client secret
```
This is a nit, so feel free to ignore if you don't like it.

> You'll also need to provide the client ID and secret from your identity provider

Reviewer 2: ```suggestion
You'll also need to provide the client ID and client secret from your identity provider
```
Again, another nit.
